instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/TimeUnitHelper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.micrometer.v1_5;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class TimeUnitHelper {
+
+  private static final Logger logger = LoggerFactory.getLogger(OpenTelemetryMeterRegistry.class);
+
+  static TimeUnit parseConfigValue(@Nullable String value) {
+    if (value == null) {
+      return TimeUnit.MILLISECONDS;
+    }
+    switch (value) {
+      case "ns":
+      case "nanoseconds":
+        return TimeUnit.NANOSECONDS;
+      case "us":
+      case "microseconds":
+        return TimeUnit.MICROSECONDS;
+      case "ms":
+      case "milliseconds":
+        return TimeUnit.MILLISECONDS;
+      case "s":
+      case "seconds":
+        return TimeUnit.SECONDS;
+      case "min":
+      case "minutes":
```

Comment:
are these names and abbreviations from micrometer/otel/ucum?

Reply:
Shorter names come from UCUM, the longer ones are just `TimeUnit` values lowercased - I thought it would be quite useful to have both